gtk4/src/custom_sorter.rs

```diff
@@ -0,0 +1,52 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::CustomSorter;
+use crate::Ordering;
+use glib::object::IsA;
+use glib::translate::*;
+
+pub trait CustomSorterExtManual: 'static {
+    fn set_sort_func<F>(&self, sort_func: F)
+    where
+        F: Fn(&glib::Object, &glib::Object) -> Ordering;
+}
+
+unsafe extern "C" fn destroy_closure<F: Fn(&glib::Object, &glib::Object) -> Ordering>(
+    ptr: glib::ffi::gpointer,
+) {
+    Box::<F>::from_raw(ptr as *mut _);
+}
+
+unsafe extern "C" fn trampoline<F: Fn(&glib::Object, &glib::Object) -> Ordering>(
+    a: glib::ffi::gconstpointer,
+    b: glib::ffi::gconstpointer,
+    f: glib::ffi::gpointer,
+) -> i32 {
+    let f: &F = &*(f as *const F);
+    f(&from_glib_none(a as *mut glib::gobject_ffi::GObject), &from_glib_none(b as *mut glib::gobject_ffi::GObject)).to_glib()
+}
+
+fn into_raw<F>(func: F) -> glib::ffi::gpointer
+where
+    F: Fn(&glib::Object, &glib::Object) -> Ordering,
+{
+    skip_assert_initialized!();
+    let func: Box<F> = Box::new(func);
+    Box::into_raw(func) as glib::ffi::gpointer
+}
+
+impl<O: IsA<CustomSorter>> CustomSorterExtManual for O {
+    fn set_sort_func<F>(&self, sort_func: F)
+    where
+        F: Fn(&glib::Object, &glib::Object) -> Ordering,
+    {
+        unsafe {
```

Comment:
I would recommend moving all those helper functions (trampoline, into raw, destroy) into this function. No need to pollute the module scope with them.

Reply:
They will be reused for the new function no?

Reply:
Now yes :) `into_raw()` is a bit useless though, just put `Box::into_raw(Box::new(func)) as *mut _` on the caller side